glommio/src/read_write_lock.rs

```diff
@@ -0,0 +1,1214 @@
+        });
+    }
+
+    #[test]
+    fn test_global_lock() {
+        test_executor!(async move {
+            let rc = Rc::new(ReadWriteLock::new(0));
+            let rc2 = rc.clone();
+
+            let s = Rc::new(Semaphore::new(0));
+            let s2 = s.clone();
+
+            let mut fibers = Vec::new();
+            fibers.push(Task::local(async move {
+                let mut lock = rc2.write().await.unwrap();
+
+                for _ in 0..10 {
+                    Task::<()>::later().await;
+                    let tmp = *lock;
+                    *lock -= 1;
+                    Task::<()>::later().await;
+                    *lock = tmp + 1;
+                }
+
+                s2.signal(1);
+            }));
+
+            for _ in 0..5 {
+                let rc3 = rc.clone();
+
+                fibers.push(Task::local(async move {
+                    let lock = rc3.read().await.unwrap();
+                    assert!(*lock == 0 || *lock == 10);
+
+                    Task::<()>::later().await;
+                }));
+            }
+
+            join_all(fibers).await;
+            s.acquire(1).await.unwrap();
+            let lock = rc.read().await.unwrap();
+            assert_eq!(*lock, 10);
+        });
+    }
+
+    #[test]
+    fn test_local_lock() {
+        test_executor!(async move {
+            let rc = Rc::new(ReadWriteLock::new(0));
+            let rc2 = rc.clone();
+
+            let s = Rc::new(Semaphore::new(0));
+            let s2 = s.clone();
+
+            let mut fibers = Vec::new();
+            fibers.push(Task::local(async move {
+                for _ in 0..10 {
+                    let mut lock = rc2.write().await.unwrap();
+                    let tmp = *lock;
+                    *lock -= 1;
+
+                    Task::<()>::later().await;
+
+                    *lock = tmp + 1;
+                }
+
+                s2.signal(1);
+            }));
+
+            for _ in 0..5 {
+                let rc3 = rc.clone();
+
+                fibers.push(Task::local(async move {
+                    let lock = rc3.read().await.unwrap();
+                    assert!(*lock >= 0);
+
+                    Task::<()>::later().await;
+                }));
+            }
+
+            join_all(fibers).await;
+            s.acquire(1).await.unwrap();
+            let lock = rc.read().await.unwrap();
+            assert_eq!(*lock, 10);
+        });
+    }
+
+    #[test]
+    fn test_ping_pong() {
+        test_executor!(async move {
+            const ITERATIONS: i32 = 10;
+
+            let ball = Rc::new(ReadWriteLock::new(0));
+            let ball2 = ball.clone();
+            let ball3 = ball.clone();
+
+            let mut fibers = Vec::new();
+
+            let pinger = Task::local(async move {
+                let mut prev = -1;
+                loop {
+                    //give a room for other fibers to participate
+                    Task::<()>::later().await;
```

Comment:
`Local` is a bit more idiomatic than `Task::<()>`

Reply:
OK